api/cmd/portainer/main.go

```diff
@@ -87,40 +88,19 @@ func initDataStore(flags *portainer.CLIFlags, fileService portainer.FileService,
 		return nil
 	}
 
-	// Init sets some defaults - its basically a migration
 	err = store.Init()
 	if err != nil {
 		log.Fatalf("failed initializing data store: %v", err)
 	}
 
-	if isNew {
-		// from MigrateData
-		store.VersionService.StoreDBVersion(portainer.DBVersion)
-
-		// EXPERIMENTAL, will only activate if `/data/import.json` exists
-		importFromJson(fileService, store)
-
-		err := updateSettingsFromFlags(store, flags)
-		if err != nil {
-			log.Fatalf("failed updating settings from flags: %v", err)
-		}
-	}
-
-	storedVersion, err := store.VersionService.DBVersion()
+	err = store.MigrateData()
```

Comment:
the isNew and decision to run a migration was moved here to make those more explicit (ie, please don't just call `store.MigrateData()` like this) as in the before refactor version, this resulted in a number of unexpected bugs.